src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMapping.java

```diff
@@ -271,10 +280,20 @@ public static List<RelatedParty> getIndividualRelatedParties(Entity entity, Stri
         return relatedParties;
     }
 
+    /**
+     * Returns related party according to given id and relationship type.
+     * According to task <a href ="https://tools.hmcts.net/jira/secure/RapidBoard.jspa?rapidView=2115&projectKey=SSCS&view=detail&selectedIssue=SSCS-10747">SSCS-10747</a>
+     * for representatives it is asked to return solicitor code as the relationship type
+     * which is 11, REPRESENTATIVE_RELATION_TYPE_SOLICITOR_CODE
+     *
+     * @param id id of the related party
+     * @param relationshipType relationship type of the related party
+     * @return RelatedParty details
+     */
     public static RelatedParty getRelatedParty(String id, String relationshipType) {
         return RelatedParty.builder()
                 .relatedPartyId(id)
-                .relationshipType(relationshipType)
+                .relationshipType(REPRESENTATIVE.getHmcReference().equals(relationshipType) ? REPRESENTATIVE_RELATION_TYPE_SOLICITOR_CODE : relationshipType)
```

Comment:
This logic shouldn't be here can you move it to getIndividualRelatedParties. There you can set the relationshipType in the switch instead of doing this.

Reply:
moved that control to getIndividualRelatedParties method.